stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,217 @@
+                        data = []
+                        break
+
+                if data:
+                    data = self.get_results_data(data)
+                    if metadata:
+                        return_obj['data'] = data if data else []
+                    else:
+                        return_obj['data'] = data[int(offset): total_records] if data else []
+                    if jwt_token and next_url and result_count < self.api_client.result_limit:
+                        return_obj['metadata'] = {"result_count": result_count, "jwtToken": jwt_token,
+                                                  "next_url": next_url}
+                else:
+                    if not return_obj.get('error') and return_obj.get('success') is not False:
+                        return_obj['success'] = True
+                        return_obj['data'] = []
+            else:
+                return_obj['success'] = True
+                return_obj['data'] = []
+
+        except InvalidMetadataException as ex:
+            return_obj = self.exception_response(422, f'Invalid metadata: {str(ex)}')
+
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: return_object, dict
+        """
+        return_obj = {}
+        try:
+            token = await self.get_token()
+            response = await self.api_client.ping_data_source(token)
+            response_code = response.code
+            response_dict = json.loads(response.read().decode('utf-8'))
+            if response_code == 200:
+                return_obj['success'] = True
+            else:
+                return_obj = self.exception_response(response_code, response_dict.get('errorSummary', ''))
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def get_token(self):
+        """ Generate new token"""
+        response = await self.api_client.generate_token()
+        response_code = response.code
+        response_txt = response.read().decode('utf-8')
+        response_json = json.loads(response_txt)
+        if response_code == 200:
+            response_header = response.headers
+            if 'Authorization' in response_header:
+                token = response_header.get('Authorization')
+            return token
+        raise Exception(response_json['errors'])
+
+    def exception_response(self, code=None, response_txt=''):
+        """
+        create the exception response
+        :param code, int
+        :param response_txt, str
+        :return: return_obj, dict
+        """
+        return_obj = {}
+        message = None
+
+        if 'Query is not valid' in response_txt:
+            code = 400
+            message = 'Query is not valid'
+
+        if code == 401 and not response_txt:
```

Comment:
This doesn't seem neccessary. You should already have the code from the API response and if the response has a message you can just use that.

Reply:
If the query is not valid from the API side, we are getting a response code of 200.
We are always keeping the code 200 as a successive response, so that's why we are changing the code to 400.